splink/postgres/postgres_linker.py

```diff
@@ -0,0 +1,246 @@
+
+        super().__init__(
+            input_tables,
+            settings_dict,
+            accepted_df_dtypes,
+            set_up_basic_logging,
+            input_table_aliases=input_aliases,
+        )
+
+    def _table_to_splink_dataframe(self, templated_name, physical_name):
+        return PostgresDataFrame(templated_name, physical_name, self)
+
+    def _execute_sql_against_backend(self, sql, templated_name, physical_name):
+        # In the case of a table already existing in the database,
+        # execute sql is only reached if the user has explicitly turned off the cache
+        self._delete_table_from_database(physical_name)
+
+        sql = f"CREATE TABLE {self._db_schema}.{physical_name} AS {sql}"
+        self._log_and_run_sql_execution(sql, templated_name, physical_name)
+
+        output_obj = self._table_to_splink_dataframe(templated_name, physical_name)
+        return output_obj
+
+    def _run_sql_execution(
+        self, final_sql: str, templated_name: str, physical_name: str
+    ) -> SplinkDataFrame:
+        cur = self.con.cursor()
+        cur.execute(final_sql)
+        self.con.commit()
+        cur.close()
+
+    def register_table(self, input, table_name, overwrite=True):
+        # If the user has provided a table name, return it as a SplinkDataframe
+        if isinstance(input, str):
+            return self._table_to_splink_dataframe(table_name, input)
+
+        # Check if table name is already in use
+        exists = self._table_exists_in_database(table_name)
+        if exists:
+            if not overwrite:
+                raise ValueError(
+                    f"Table '{table_name}' already exists in database. "
+                    "Please use the 'overwrite' argument if you wish to overwrite"
+                )
+            else:
+                self._delete_table_from_database(table_name)
+
+        if isinstance(input, dict):
+            input = pd.DataFrame(input)
+        elif isinstance(input, list):
+            input = pd.DataFrame.from_records(input)
+
+        def connection_fairy():
+            return self.con
+
+        engine = create_engine("postgresql://", creator=connection_fairy)
+
+        # Will error if an invalid data type is passed
+        input.to_sql(
+            table_name,
+            con=engine,
+            index=False,
+            if_exists="replace",
+            schema=self._db_schema,
+        )
+        return self._table_to_splink_dataframe(table_name, table_name)
+
+    def _random_sample_sql(self, proportion, sample_size, seed=None):
+        if proportion == 1.0:
+            return ""
+        if seed is None:
+            seed = "random()"
+
+        sample_size = int(sample_size)
+        return f"""
+                WHERE unique_id IN (
+                    SELECT unique_id
+                    FROM __splink__df_concat_with_tf"
+                    ORDER BY {seed}
+                    LIMIT {sample_size}
+                )
+            """
```

Comment:
Perhaps I am missing something, but I'm not sure if this gives a properly random sample if `seed` is provided? Would it be possible to use `TABLESAMPLE` and `REPEATABLE` or was there some reason for this other approach?

Reply:
(additionally there is a stray " after `__splink__df_concat_with_tf`)